### PR TITLE
Add support for the new NL interface

### PR DESCRIPTION
**Pull request by blegat**
Closes https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/issues/160

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -246,15 +246,73 @@ function parser_MOI(moimodel, index_map)
...
+    )
+  end
+  return dest
+end
+
+
+function _nlp_model(model::MOI.ModelLike)
+  nlp_model = nothing
+  for (F, S) in MOI.get(model, MOI.ListOfConstraintTypesPresent())
+    nlp_model = _nlp_model!(model, F, S)
```
Comment (abelsiqueira) [inline src/utils.jl:275]:
```suggestion
    nlp_model = _nlp_model(model, F, S)
```

**Comment by abelsiqueira** [comment]
Thanks for the PR. I looked in the logs and found this small type. Maybe that is the only issue, but to be honest it's way beyond my pay grade.
It would be nice to have a test with JuMP 1.15 nonlinear syntax as well if you have the time.

**Comment by blegat** [comment]
Because we run `MOI.Test.runtests` and we now declare that we support `ScalarNonlinearFunction`, it should automatically now run with tests in `MOI.ScalarNonlinearFunction`

**Review by abelsiqueira — APPROVED** [review]
Thanks!

**Comment by abelsiqueira** [comment]
Release 0.12.5 includes this.

**Comment by tmigot** [comment]
Thanks @blegat and @abelsiqueira that looks great!

**Pull request merged**